applications/MappingApplication/custom_utilities/interface_vector_container.h

```diff
@@ -0,0 +1,136 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher, Jordi Cotela
+//
+// See Master-Thesis P.Bucher
+// "Development and Implementation of a Parallel
+//  Framework for Non-Matching Grid Mapping"
+
+#if !defined(KRATOS_INTERFACE_VECTOR_CONTAINER_H_INCLUDED )
+#define  KRATOS_INTERFACE_VECTOR_CONTAINER_H_INCLUDED
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "mapper_utilities.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+ */
+template<class TSparseSpace, class TDenseSpace>
+class InterfaceVectorContainer
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of InterfaceVectorContainer
+    KRATOS_CLASS_POINTER_DEFINITION(InterfaceVectorContainer);
+
+    typedef typename TSparseSpace::VectorType TSystemVectorType;
+
+    typedef Kratos::unique_ptr<TSystemVectorType> TSystemVectorUniquePointerType;
+
+    typedef Variable<double> DoubleVariableType;
+    typedef VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > ComponentVariableType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    InterfaceVectorContainer(ModelPart& rModelPart) : mrModelPart(rModelPart) {}
+
+    /// Destructor.
+    virtual ~InterfaceVectorContainer() = default;
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void UpdateSystemVectorFromModelPart(const DoubleVariableType& rVariable,
+                                         const Kratos::Flags& rMappingOptions);
+
+    void UpdateSystemVectorFromModelPart(const ComponentVariableType& rVariable,
+                                         const Kratos::Flags& rMappingOptions);
+
+    void UpdateModelPartFromSystemVector(const DoubleVariableType& rVariable,
+                                         const Kratos::Flags& rMappingOptions);
+
+    void UpdateModelPartFromSystemVector(const ComponentVariableType& rVariable,
+                                         const Kratos::Flags& rMappingOptions);
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    TSystemVectorType& GetVector()
+    {
+        KRATOS_DEBUG_ERROR_IF_NOT(mpInterfaceVector)
+            << "The Interface-Vector was not initialized" << std::endl;
+        return *mpInterfaceVector;
+    }
+
+    TSystemVectorUniquePointerType& pGetVector()
```

Comment:
The class would be more complete with const overloads. Or, since you provide accessors for all the private members, you can just make it a struct and expose them.

Reply:
I added the const overloads
I will leave it as a class since I plan to slightly enhance the access in the future (maybe using trilinos-multivector)